src/core/service.c

```diff
@@ -1641,6 +1641,17 @@ static int service_spawn_internal(
                 }
         }
 
+        Job *j = UNIT(s)->job;
+        if (j && j->activation_event_info) {
```

Comment:
so, this really should read the value off `u`, not of `u->job`...

It's indication that the copying from job to unit happens too late, as discussed above: plesae pass the info from job to unit as part of unit_start() parameters, so that it is in place here already, and you can just pick it up from `u->job`.